src/wiktextract/extractor/simple/preprocess.py

```diff
@@ -0,0 +1,113 @@
+from .parse_utils import PANEL_TEMPLATES
+from .section_titles import POS_DATA
+from .text_utils import (
+    HEADING_RE,
+    POS_STARTS_RE,
+    POS_TEMPLATE_NAMES,
+    SECTION_STARTERS,
+    TEMPLATE_NAME_RE,
+)
+
+
+def preprocess_text(text: str, page_title="") -> str:
```

Comment:
IMO code here changes wikitext is too complicated and risky, it needs at least some tests to let others (and yourself a few weeks later) to know what's it does...

I'm not fully aware the page layout this function wants to change, but I think the code will be easier to maintain if this process step could be avoided.

I guess the problem is section "Word parts" and "Pronunciation" uses level 3 node and  placed above the POS level 2 node? How about extract the level 3 section data to the base `WordEntry` data, then it will be shared to all following POS sections. For example:

```python
def parse_page(
    wxr: WiktextractContext, page_title: str, page_text: str
) -> list[dict[str, Any]]:
    wxr.wtp.start_page(page_title)
    tree = wxr.wtp.parse(page_text)
    page_data = []
    base_data = WordEntry()
    for level_node in tree.find_child(LEVEL_KIND_FLAGS):
        title_text = clean_node(wxr, None, level_node.largs)
        if title_text is pos:
            new_data = base_data.model_copy(deep=True)
            extract_pos(wxr, level_node, new_data)
            page_data.append(new_data)
        elif title_text == "Word parts":
            extract_word_parts(wxr, level_node, base_data)
        elif title_text == "Pronunciation":
            extract_pronunciation(wxr, level_node, base_data)
    return [m.model_dump(exclude_defaults=True) for m in page_data]
```

The current `parse_page()` could be simplified IMO.

Reply:
In retrospect this might be true, I'll take a look at it. You start of solving one problem, and then things escalate... There were many steps that led to what you see in the code.